cli/src/pcluster/models/cluster.py

```diff
@@ -175,17 +175,18 @@ def _get_unretained_cw_log_group_resource_keys(self):
 
     def _persist_stack_resources(self, keys):
         """Set the resources in template identified by keys to have a DeletionPolicy of 'Retain'."""
+        template = self.template
```

Comment:
Why not using self.template directly?

Reply:
Because self.template returns `yaml.safe_load(AWSApi.instance().cfn.get_stack_template(self.name))`. It will always requery the exisiting stack info, although we modify the DeletionPocliy here, we will get original template by using self.template. But we actually need modified template to update the stack.